CBLClient/Apps/CBLTestServer-Java-Desktop/src/main/java/com/couchbase/mobiletestkit/javatestserver/TestServerMain.java

```diff
@@ -0,0 +1,34 @@
+package com.couchbase.mobiletestkit.javatestserver;
+
+import com.couchbase.mobiletestkit.javacommon.Context;
+import com.couchbase.mobiletestkit.javalistener.Server;
+import com.couchbase.mobiletestkit.javacommon.util.Log;
+import com.couchbase.lite.CouchbaseLite;
+import org.nanohttpd.protocols.http.NanoHTTPD;
+import org.nanohttpd.util.ServerRunner;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+
+public class TestServerMain {
+    private static final int PORT = 8080;
+    private static Context context;
+
+    public static void main(String[] args) throws IOException {
+        Log.init(new TestServerLogger());
+        CouchbaseLite.init();
+
+        String curPath = new File(".").getCanonicalPath();
+        context = new TestServerContext(new File(curPath));
+        Server.setContext(context);
+
+        String ip = context.getLocalIpAddress();
+        Server.memory.setIpAddress(ip);
```

Comment:
You might want to bind to 0.0.0.0 here, instead of looking up the ip addy.  I believe that would bind to localhost, too...

Reply:
you are right, I have cleaned up the context.getLocalIpAddress() implementation, and simply let this method to return a string of 0.0.0.0. Keeping getLocalIpAddress() method definition because it's in interface, android has it own copy of implementation to track ip on device